parlai/core/worlds.py

```diff
@@ -550,6 +550,13 @@ def __next__(self):
 
     def batch_observe(self, index, batch_actions, index_acting):
         batch_observations = []
+        a = self.world.get_agents()[index]
+
+        if (batch_actions is not None and len(batch_actions) > 0 and
+                hasattr(a, 'batch_observe')) and index != index_acting:
+            # if there's anything that needs to be batched, do it here
+            a.batch_observe(batch_actions)
```

Comment:
@jaseweston @alexholdenmiller Thanks for the comments! `w.observe` is an easy fix.

However, the point of keeping both `batch_observe` and `observe` (instead of making them mutually exclusive) is so that task agents (`DialogTeacher`s) can directly inherit the nice behavior of metric updates from the base class' `act` and `observe` methods. Would be happy to make them mutually exclusive if we find out a nice way of inheriting those...